src/collection/pages/main/main.tsx

```diff
@@ -278,10 +273,10 @@ const Main: React.FC = () => {
       });
       return collectionResult.data!.action!.monsterCollect as string;
     } else if (data.stateQuery.agent!.monsterCollectionLevel > latestCollectionItem.tier) {
-      const collectionResult = await cancelCollection({
-        variables: { level: latestCollectionItem.tier },
+      const collectionResult = await collect({
+        variables: { level: 0 },
```

Comment:
This logic is unnecessary unless `cancelCollect` and `collect` are separate. It seems better to fix the logic to do a `collect` only when it is not 0.

Reply:
Amended. https://github.com/planetarium/9c-launcher/pull/788/commits/5e35dfa5afcc96ea53e9ef8181520c270a3c81e5